rpcs3/Emu/Cell/PPUInterpreter.cpp

```diff
@@ -980,22 +958,28 @@ bool ppu_interpreter::VLOGEFP(ppu_thread& ppu, ppu_opcode_t op)
 
 bool ppu_interpreter_fast::VMADDFP(ppu_thread& ppu, ppu_opcode_t op)
 {
+	// Set FTZ (Flush denormals to zero) to 1, fix issue #6296
+	_MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
 	const auto a = ppu.vr[op.va].vf;
 	const auto b = ppu.vr[op.vb].vf;
 	const auto c = ppu.vr[op.vc].vf;
-	auto result = _mm_add_ps(_mm_mul_ps(a, c), b);	
-	ppu.vr[op.vd] = vec_handle_nan(vec_handle_subnormal(result));
+	const auto result = _mm_add_ps(_mm_mul_ps(a, c), b);
+	ppu.vr[op.vd] = vec_handle_nan(result);
+	// Restore FTZ
+	_MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_OFF);
 	return true;
 }
 
 bool ppu_interpreter_precise::VMADDFP(ppu_thread& ppu, ppu_opcode_t op)
 {
+	// Set FTZ (Flush denormals to zero) to 1, fix issue #6296
+	_MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
 	const auto a = ppu.vr[op.va];
 	const auto b = ppu.vr[op.vb];
 	const auto c = ppu.vr[op.vc];
-	auto result = v128::fma32f(a, c, b);
-	result.vf    = vec_handle_subnormal(result.vf);
-	ppu.vr[op.rd] = vec_handle_nan(result, a, b, c);
+	ppu.vr[op.rd] = vec_handle_nan(v128::fma32f(a, c, b), a, b, c);
+	// Restore FTZ
+	_MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_OFF);
 	return true;
```

Comment:
Sadly, this isn't a very good solution, the intel optimization manual mentions that the FTZ/DAZ flags are slow to set, and shouldn't really be used in this way:

From page 170 of https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-optimization-manual.pdf:

> It is expected that SIMD applications are unlikely to alternate between FTZ and DAZ mode values. Consequently, the SIMD control word does not have the short latencies that the floating-point control register does. A read of the MXCSR register has a fairly long latency, and a write to the register is a seri-alizing instruction.